src/game/Game.kt

```diff
@@ -2,31 +2,39 @@ package game
 
 import action.Action
 import agent.Player
+import map.AlwaysGoUpPlayer
 import map.ExitNode
 import map.WorldMap
 import java.lang.UnsupportedOperationException
 
 class Game(private val player: Player, private val worldMap: WorldMap) {
+    companion object {
+        const val WALL_FEEDBACK = "You cannot go forward, as you have hit a wall."
+    }
 
     var timeCounter: Int = 0
 
     fun run() {
         while(!Thread.interrupted()) {
-            val nextMove = player.chooseNextMove()
-            when (nextMove) {
-                Action.MOVE_NORTH ->
-                    worldMap.moveObject(player.actor, 0, -1)
-                else -> {
-                    throw UnsupportedOperationException()
-                }
+            when (player.chooseNextMove()) {
+                Action.MOVE_NORTH -> moveNorthLogic()
             }
-
-            if (!player.actor.alive || worldMap.getNode(worldMap.positionFor(player.actor)) is ExitNode) {
+            if (!player.actor.alive ||
+                    player.feedback == WALL_FEEDBACK ||
```

Comment:
I dont think the game should stop if the player tries to go through the wall. Just players position doesnt change, he gets the feedback, but the world goes on. Player should have a chance to correct his mistake (ie learn the concept of the wall on his own).
If you do implement it this way, however, you will have a problem of infinite game, but we could address this by adding a more generic terminate condition (global turn limit or player not moving for x turns etc)

Reply:
you're right will terminate after player is not alive anymore - to achieve this he has to lose 1 life unit per turn